Userland/Libraries/LibGfx/QOILoader.cpp

```diff
@@ -169,23 +168,21 @@ static ErrorOr<NonnullRefPtr<Bitmap>> decode_qoi_image(InputMemoryStream& stream
                 if (stream.handle_any_error())
                     return Error::from_string_literal("Invalid QOI image: end of stream while reading chunk tag"sv);
                 if (tag == QOI_OP_RGB)
-                    pixel = TRY(decode_qoi_op_rgb(stream, pixel));
```

Comment:
It's not really clear that encapsulating this logic in a class for this single case is a useful abstraction. I personally think it hurts readability and adds unnecessary complexity to the code.

Reply:
I now see this is used as a base class in later commits (was reviewing commit by commit). Adding such details to the commit message would valuable to the reviewer.